core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -207,8 +191,16 @@ public void runInContext() {
       });
       return;
     }
-    prepareHeaders(headers, callOptions, userAgent,
-        knownMessageEncodingRegistry, decompressorRegistry, compressorRegistry);
+    String compressorName = callOptions.getCompressor();
+    if (compressorName != null) {
+      compressor = compressorRegistry.lookupCompressor(compressorName);
+      checkArgument(compressor != null, "Unable to find compressor by name %s", compressorName);
```

Comment:
I'm a bit unsettled by the exception given that start is not documented to throw an exception in this case and how rarely we throw exceptions from our API. Would it be better to call `onClose` with a status instead? (maybe something similar to what is done for `context.isCancelled()` above)